### PR TITLE
upgrade to latest sdk

### LINKED ISSUE #201: Upgrade to latest stable Stargate Cosmos SDK release
- [ ] Update stakewatcher
- [ ] Update stakebot

### PR DESCRIPTION
**Pull request by jhernandezb**
Fixes #201

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -14,14 +14,15 @@ steps:
...
-      - golangci-lint run -v --timeout 3m --tests=false
+      - golangci-lint run -v --timeout 5m --tests=false
     environment:
       GOPROXY: http://goproxy
   - name: test-and-build
     image: golang:1.15.2
     commands:
       - apt-get update && apt-get install -y cmake
       - ./contrib/go-test.sh
+      - FAUCET_ENABLED=true make build-linux
```
Comment (shanev) [inline .drone.yml:25]: 🤩

File: Makefile [hunk 2]
```diff
@@ -1,40 +1,108 @@
...
 
-BUILD_FLAGS := -ldflags '$(ldflags)'
 
 all: install
 
 create-wallet:
-	stakecli config keyring-backend test
-	stakecli config chain-id localnet-1
-	stakecli keys add validator
+	staked keys add validator --keyring-backend test
```
Comment (shanev) [inline Makefile:82]: No longer need `chain-id` here?

File: proto/stakebird/curating/v1beta1/query.proto [hunk 3]
```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
```
Comment (shanev) [inline proto/stakebird/curating/v1beta1/query.proto:1]: This is cool. Can possibly be used for Stakebot resolvers too.

File: x/curating/abci_test.go [hunk 4]
```diff
@@ -8,19 +8,21 @@ import (
...
-	_, app, ctx := testdata.CreateTestInput()
+func setup(t *testing.T) (*simapp.SimApp, sdk.Context) {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
 
 	postID := "500"
 	vendorID := uint32(1)
 
-	addrs = testdata.AddTestAddrsIncremental(app, ctx, 3, sdk.NewInt(10_000_000))
+	addrs = simapp.AddTestAddrsIncremental(app, ctx, 3, sdk.NewInt(10_000_000))
```
Comment (shanev) [inline x/curating/abci_test.go:25]: Nice!

File: x/curating/keeper/post.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ func (k Keeper) GetPostZ(
 	if value == nil {
 		return post, false, nil
 	}
-	k.cdc.MustUnmarshalBinaryBare(value, &post)
+	k.MustUnmarshalPost(value, &post)
```
Comment (shanev) [inline x/curating/keeper/post.go:35]: So much nicer.

**Review by shanev — APPROVED** [review]
Thanks for this. I know it was a massive effort. Lots of simplifications, app.go is finally broken up into manageable pieces, and nice work in using simapp over testdata.

🤩🥳🦄

File: Makefile (hunk 2 above)
Comment (jhernandezb) [reply Makefile:82]: Good point I guess I still have to figure out how to wire `stakecli config ...` in staked

Comment (shanev) [reply Makefile:82]: No worries if it's not there yet, but pass in the `chain-id` if this command needs it.

Comment (jhernandezb) [reply Makefile:82]: This particular one doesn't needs it, but interacting with transactions will require a explicit --chain-id.

**Review by JakeHartnell — APPROVED** [review]
Amazing work! Was able to create a new post and upvote, so this is ready! :heart: :rocket:

**Pull request merged**